rasa/cli/train.py

```diff
@@ -59,6 +59,8 @@ def train(args: argparse.Namespace) -> Optional[Text]:
         for f in args.data
     ]
 
+    validate_domain(domain)
+
```

Comment:
Do we need these in the CLI functions? This calls `rasa.train` which will initialize an agent (which will validate the domain). I think this is the actual cause of the double logging

Reply:
The idea I had is to fail as early as possible. So, all entry points that need a domain, should validate the domain before using it. Will remove the sanity check in the agent again to avoid double logging.

Reply:
I definitely understand that, but wonder if doing that is worth separating the functionality from its class (e.g. I feel like all domain validation stuff should stay as close to the domain as possible). does that make sense? maybe this would also change when we introduce the validator class